synapse/models/infotech.py

```diff
@@ -71,9 +65,211 @@ def getBaseModels():
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('exe', {'ptype': 'file:bytes', 'doc': 'The file containing the code that read the file'}),
+
+                )),
+
+                ('it:exec:file:write', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('path', {'ptype': 'file:path'}),
+                    ('path:dir', {'ptype': 'file:path', 'ro': 1}),
+                    ('path:ext', {'ptype': 'str:lwr', 'ro': 1}),
+                    ('path:base', {'ptype': 'file:base', 'ro': 1}),
+                    ('file', {'ptype': 'file:bytes'}),
+                    ('time', {'ptype': 'time'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('exe', {'ptype': 'file:bytes', 'doc': 'The file containing the code that wrote to the file'}),
+
+                )),
+
+                ('it:exec:reg:get', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
+
+                ('it:exec:reg:set', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
+
+                ('it:exec:reg:del', {'ptype': 'guid'}, (
+                    ('host', {'ptype': 'it:host'}),
+                    ('reg', {'ptype': 'it:dev:regval'}),
+                    ('reg:key', {'ptype': 'it:dev:regkey', 'ro': 1}),
+                    ('reg:int', {'ptype': 'it:dev:int', 'ro': 1}),
+                    ('reg:str', {'ptype': 'it:dev:str', 'ro': 1}),
+                    ('reg:bytes', {'ptype': 'file:bytes', 'ro': 1}),
+                    ('exe', {'ptype': 'file:bytes'}),
+                    ('proc', {'ptype': 'it:exec:proc'}),
+                    ('time', {'ptype': 'time'}),
+                )),
             ),
 
         }
-        name = 'it'
-        return ((name, modl), )
+
+        models = (
+            ('it', modl),
+        )
+
+        return models
+
+    def intiCoreModule(self):
```

Comment:
typo???

Reply:
:+1:

Reply:
nope.  the base model no longer uses this for anything, but provides it for sub-class implementations to implement if they want.

Reply:
intiCoreModule is misspelled